sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs

```diff
@@ -43,21 +63,33 @@ impl Symbol {
 
     fn bitbanging_byte<const SDA: u8, const SCL: u8>(byte: u8, nack: bool, samples: &mut Vec<u8>) {
         let data: u16 = (byte as u16) << 1u16 | nack as u16;
-        samples.extend((0..9u8).rev().flat_map(|bit| {
-            [
+        let mut prev_scl: u16 = (samples.last().unwrap_or(&(0x01 << SCL)) & (0x01 << SCL)) as u16;
```

Comment:
You don't need the last clock, similarly to the previous version, you can assume that the clock will always be left low by the last function that operated on samples. This makes the code simpler. And then you can keep the `flat_map`.